nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -561,19 +556,32 @@ def forward_internal(
             pad_mask = pad_mask[:, cache_len:]
             if att_mask is not None:
                 att_mask = att_mask[:, cache_len:]
+            # Convert caches from the tensor to list
+            cache_last_time_next = [None] * self.streaming_cfg.last_time_num
+            cache_last_channel_next = [None] * self.streaming_cfg.last_channel_num
 
         for lth, (drop_prob, layer) in enumerate(zip(self.layer_drop_probs, self.layers)):
             original_signal = audio_signal
+            if cache_last_channel is not None:
+                cache_last_channel_cur = cache_last_channel[layer.self_attn._cache_id]
```

Comment:
With the latest changes, I think there is no need for cache_id and all these stuff anymore. You may just call each layer and just append the next_cache returned in a list. You may pass something like cache_last_channel[lth] to each layer.